src/parser.js

```diff
@@ -293,14 +296,231 @@ export default class Parser {
+
+        if (this.generateSourceMap) {
+          this.createSourceMap(scope.meta.get('description'), this.path.concat([scopeName]));
+        }
+      }
+
+      if (this.generateSourceMap) {
+        const value = descriptions ? scopeName : index;
+        this.createSourceMap(scope, this.path.concat([value]));
+      }
+
+      scopes.content.push(scope);
+    });
+
+    if (scopes.length) {
+      element.content.push(new MemberElement('scopes', scopes));
+    }
+  }
+
+  // Conver OAuth2 transition information into Refract elements
+  oauthTransitions(element, oauth) {
+    const {Transition} = this.minim.elements;
+
+    if (oauth.authorizationUrl) {
+      const transition = new Transition();
+
+      transition.relation = 'authorize';
+      transition.href = oauth.authorizationUrl;
+
+      if (this.generateSourceMap) {
+        this.createSourceMap(transition.attributes.get('href'), this.path.concat(['authorizationUrl']));
+        this.createSourceMap(transition.attributes.get('relation'), this.path.concat(['authorizationUrl']));
+      }
+
+      element.content.push(transition);
+    }
+
+    if (oauth.tokenUrl) {
+      const transition = new Transition();
+
+      transition.relation = 'token';
+      transition.href = oauth.tokenUrl;
+
+      if (this.generateSourceMap) {
+        this.createSourceMap(transition.attributes.get('href'), this.path.concat(['tokenUrl']));
+        this.createSourceMap(transition.attributes.get('relation'), this.path.concat(['tokenUrl']));
+      }
+
+      element.content.push(transition);
+    }
+  }
+
   // Convert a Swagger auth object into Refract elements.
   handleSwaggerAuth() {
-    for (const attribute of ['securityDefinitions', 'security']) {
-      if (this.swagger[attribute]) {
-        this.createAnnotation(annotations.DATA_LOST, [attribute],
-          'Authentication information is not yet supported');
-      }
+    const {Category, AuthScheme} = this.minim.elements;
+    const schemes = [];
+
+    if (this.swagger.securityDefinitions) {
+      _.each(_.keys(this.swagger.securityDefinitions), (name) => {
+        this.withPath('securityDefinitions', name, () => {
+          const item = this.swagger.securityDefinitions[name];
+          const element = new AuthScheme();
+
+          switch (item.type) {
+          case 'basic':
+            element.element = 'Basic Authentication Scheme';
+            break;
+
+          case 'apiKey':
+            element.element = 'Token Authentication Scheme';
+            this.apiKeyName(element, item);
+            break;
+
+          case 'oauth2':
+            element.element = 'OAuth2 Scheme';
+            this.oauthGrantType(element, item.flow);
+
+            if (item.scopes) {
+              this.withPath('scopes', () => {
+                this.oauthScopes(element, item.scopes);
+              });
+            }
+
+            this.oauthTransitions(element, item);
+            break;
+
+          default:
+            break;
```

Comment:
I think we should if we're given security definitions we don't understand. Informing the user that this scheme is not supported?